lib/fetch/request.js

```diff
@@ -418,22 +418,28 @@ class Request {
 
     // 32. If init is not empty, then:
     if (initHasKey) {
+      /** @type {HeadersList} */
       const headersList = this[kHeaders][kHeadersList]
       // 1. Let headers be a copy of this’s headers and its associated header
       // list.
       // 2. If init["headers"] exists, then set headers to init["headers"].
-      // Note: Avoid call `HeadersList` if init["headers"] is not undefined
-      const headers = init.headers !== undefined ? init.headers : new HeadersList(headersList)
+      let headers
+      if (init.headers !== undefined) {
+        headers = init.headers
+      } else {
+        init.headers = new HeadersList(headersList)
+      }
 
       // 3. Empty this’s headers’s header list.
-      // Note: Avoid call `HeadersList#clear` if already empty
-      if (headersList.mapSize !== 0) headersList.clear()
+      headersList.clear()
```

Comment:
Does  this actaully make a difference?

Reply:
Checking the Map size and not clearing is twenty times faster if the map is empty.
However, to lower maintenance costs, this change has been reverted.
If you want it back, I will gladly undo it.